### DISCUSSION TITLE
Compatibility with EXPANSION UNITS

### DISCUSSION THREAD

**Discussion by wisepds** [007revad/Synology_HDD_db#124] [Ideas]
Hello!
I have a DS3622xs+ and a DX1222 expansion unit.
Script works fine in the nas but not in the expansion unit.
Same disk inside NAS=Gren, all ok. Inside expansion unit=Red, not compatible.
Can you add this compatibility to the script please?
Thank you very much!!!

**Comment by wisepds** [comment]
By the way.. i have donated you by paypal ;)

**Comment by 007revad** [comment]
Can you run the script again and paste the output here.

**Comment by wisepds** [comment]
Here is!
![Result](https://github.com/007revad/Synology_HDD_db/assets/24432252/b9786eac-9825-449a-86d7-a1dc7642aecd)

**Comment by wisepds** [comment]
Here is the Nas with expansion unit:
![Configuration](https://github.com/007revad/Synology_HDD_db/assets/24432252/5c1b7e2f-156f-4fe9-aeae-59ff690ab9ef)

**Reply by wisepds** [reply]
I think this is the problem...
![No expansion unit](https://github.com/007revad/Synology_HDD_db/assets/24432252/e275a4d2-5091-44b1-abcc-92b4d8ca28f7)

**Reply by wisepds** [reply]
Here is in text... 


wisepds@Cloudata:/volume1/CLOUDATA/CLOUDATA/BUROCRACIA/BASEDEDATOSHDD$ sudo bash -i syno_hdd_db.sh -nr
Synology_HDD_db v3.1.61
DS3622xs+ DSM 7.2-64570
Using options: -nr

HDD/SSD models found: 1
HAT5300-16T,1403

M.2 drive models found: 1
SNV3500-400G,ECEM12.4

M.2 PCIe card models found: 1
M2D20

No Expansion Units found

HAT5300-16T already exists in ds3622xs+_host_v7.db
SNV3500-400G already exists in ds3622xs+_host_v7.db
SNV3500-400G already exists in ds3622xs+_m2d20_v7.db
SNV3500-400G already exists in ds3622xs+_m2d20_v7.db.new

M2D20 NVMe already enabled for DS3622xs+

Support disk compatibility already enabled.

Disabled support memory compatibility.

Max memory is set to 48 GB.

M.2 volume support already enabled.

Disabled drive db auto updates.

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.

**Comment by 007revad** [comment]
I notice a couple of things:
1. You are using Synology HDDs and NVMe drives, so you don't need this script.
2. Your storage manager screenshot shows the DX1222 as empty.

Do you have drives installed in the DX1222?

This script only finds expansion units if they have drives in them.

**Reply by wisepds** [reply]
Yes, i know...that's because the Exos 20TB from seagate it's outside right now and there are 3 more in the way.... 
If you want i can introduce one unit to test and you can see as a red device..
Tell me, what I do?
Thanks for your help.

**Reply by wisepds** [reply]
Ok, give me 2 seconds and i'm going to do the screen capture with unit inside.

**Reply by wisepds** [reply]
Here are screenshot with a 20TB seagate EXOS inside.
![3](https://github.com/007revad/Synology_HDD_db/assets/24432252/1dbd9c75-a338-4c95-93dc-0efecc515567)
![1](https://github.com/007revad/Synology_HDD_db/assets/24432252/31afb167-2b5c-4ca0-bda1-774121c08bbe)
![2](https://github.com/007revad/Synology_HDD_db/assets/24432252/5dd6bf3d-4e48-44a0-9fc5-cb47af3f9bc9)

**Reply by wisepds** [reply]
I have create a new volume too but nothing.. doesn't work. BUT! if i put this same disk inside NAS... VOILA! Green!!!!
![4](https://github.com/007revad/Synology_HDD_db/assets/24432252/6d847af9-39d8-4585-9a40-1aea2379d0e3)

**Comment by 007revad** [comment]
Can you run the following and report back what it outputs:
```
path="/var/log/diskprediction"
file=$(ls $path | tail -n1)
eunitlist=($(grep -Eowi "([FRD]XD?[0-9]{3,4})(rp|ii|sas){0,2}" "$path/$file" | uniq))
echo "$eunitlist"
```
Then reboot the Synology and run the above commands again and report back what it outputs:

**Reply by wisepds** [reply]
Ok, but i'll do it this night.. now there are people working on the nas, i can't reboot right now

**Comment by 007revad** [comment]
If you run it once and it returns something you don't need to reboot or run it again.

Hopefully it will return DX1222 or DX1222-1

**Reply by wisepds** [reply]
Ok, but now i'm eating with my parents.. hehehe..   in a few hours i'll do it...
Thank you again for you help!!!!

**Comment by wisepds** [comment]
This is the result:
![4](https://github.com/007revad/Synology_HDD_db/assets/24432252/31d4b997-1c3e-4b2a-8ddc-ca32bbedfe46)

**Reply by wisepds** [reply]
I think i must introduce all this instruction inside a script.. right?

**Reply by wisepds** [reply]
I introduce your instruction inside a sh script and executed it. This is the result. I don't know if is the right way to do that. No results.
![5](https://github.com/007revad/Synology_HDD_db/assets/24432252/2a576aea-e8be-4f7f-a394-765c80d0dbe8)

**Comment by 007revad** [comment]
Can you try this command:
`sudo synodiskport -enum_port -eunit`

**Reply by wisepds** [reply]
of course

**Reply by wisepds** [reply]
wisepds@Cloudata:/volume1/CLOUDATA/CLOUDATA/BUROCRACIA/BASEDEDATOSHDD$ sudo synodiskport -enum_port -eunit
Password:
 sdm sdn sdo sdp sdq sdr sds sdt

**Reply by wisepds** [reply]
The DX1222 case is conected by a special cable called Mini SAS HD (48Gbps), i don't know if this information is helpful to you...
![Minisas](https://github.com/007revad/Synology_HDD_db/assets/24432252/c1d237da-3935-453b-ba9f-918f9a875870)

**Comment by 007revad** [comment]
Try this command:
`for f in /var/log/diskprediction/*; do grep -Eowi DX1222 "$f"; done`

**Comment by wisepds** [comment]
Here is the result.
![1](https://github.com/007revad/Synology_HDD_db/assets/24432252/a68c07ac-c68e-4d81-942c-cdc08ce5b51e)

**Reply by wisepds** [reply]
@007revad Here in Spain is 1:30 AM and i'm goint to sleep cause i'm very tired. I'll read you tomorrow.
I want to help you to fix this.
Thanks again for your amazing work!

**Comment by 007revad** [comment]
Now we know that DX1222 exists in one of the diskprediction log files try this:
`for f in /var/log/diskprediction/*; do grep -Eowi "([FRD]XD?[0-9]{3,4})(rp|ii|sas){0,2}" "$f"; done`

**Reply by wisepds** [reply]
Here is the result:
![555](https://github.com/007revad/Synology_HDD_db/assets/24432252/a7cb619b-5ada-4983-bfe3-4d084bcc61cf)

**Comment by 007revad** [comment]
I've figured out what was happening.
- The script reads the latest log in /var/log/diskprediction/ to find any connected expansion units.
- But DSM only creates a disk prediction log at midnight.
- So when you ran that last command at 1:10am it found the DX12222 in the latest log.

If you run the script again it should find the DX12222 now and add your drives to the dx12222's compatible drive database.

I'll update the script to make DSM create a new disk prediction log each time the script is run.

**Comment by wisepds** [comment]
OUH YEAH!
You are the best!!!!!! :)
You'r right!! Now it's in green!!!

![q](https://github.com/007revad/Synology_HDD_db/assets/24432252/c09fc1cd-9a72-42b7-953f-b60e187a5165)

**Comment by wisepds** [comment]
I have seen the 3.1.62 version... it's 100% fixed?
Greats News!!!!!
Thank you very much @007revad
